core/src/main/scala/org/apache/spark/scheduler/cluster/StandaloneSchedulerBackend.scala

```diff
@@ -153,7 +153,7 @@ private[spark] class StandaloneSchedulerBackend(
   override def executorRemoved(fullId: String, message: String, exitStatus: Option[Int]) {
     val reason: ExecutorLossReason = exitStatus match {
       case Some(code) => ExecutorExited(code, exitCausedByApp = true, message)
-      case None => SlaveLost(message)
+      case None => SlaveLost(message, workerLost = true /* worker loss event from master */)
```

Comment:
This assumes that `exitStatus == None` implies that a worker was lost, but there are some corner-cases where this isn't necessarily true (e.g. if an executor kill fails). Looking through both the 1.6.x and 2.0.x code, it appears that `ExecutorStatus.LOST` is used exclusively for denoting whole-worker-loss, so I think that we should check that status here instead of assuming `true`. Other than that minor corner-case, this patch looks good to me, so I'll merge once we fix this.